### PR TITLE
[edn/dv] dv doc, testplan, checklist, regression, csrng_agent mod

### PR DESCRIPTION
**Pull request by senelson7**
Signed-off-by: Steve Nelson <steve.nelson@wdc.com>

### CODE REVIEW

**Comment by senelson7** [comment]
Just want @mwbranstad to review the testplan at this time. Once we finalize it, then I'll remove "draft" status.

File: hw/ip/edn/data/edn_testplan.hjson [hunk 1]
```diff
@@ -36,17 +34,18 @@
       name: csrng_cmd
       desc: '''
             Verify endpoint agent reqs generate csrng commands as predicted.
-            Verify auto-generate mode behaves as predicted.
+            Verify auto-request mode (RESEED_CMD/GENERATE_CMD registers) behaves as predicted.
             Verify max_num_reqs_between_reseeds in auto-generate mode.
```
Comment (mwbranstad) [inline hw/ip/edn/data/edn_testplan.hjson:38]: Verify that FIFO commands work with and without additional data.

File: hw/ip/edn/data/edn_testplan.hjson [hunk 2]
```diff
@@ -5,29 +5,27 @@
...
       desc: '''
-            Smoke test acessing a major datapath within the edn.
-
-            **Stimulus**:
-            - TBD
-
-            **Checks**:
-            - TBD
+            Enable edn, let edn boot mode generate csrng instantiate/generate commands,
+	    endpoint requests genbits, compare csrng/endpoint genbits.
```
Comment (mwbranstad) [inline hw/ip/edn/data/edn_testplan.hjson:16]: Verify that the software port works properly, polling each command that gets issued for ack.

**Review by mwbranstad — APPROVED** [review]
Added a few test plan items, but LGTM.

File: hw/ip/edn/data/edn_testplan.hjson (hunk 2 above)
Comment (senelson7) [reply hw/ip/edn/data/edn_testplan.hjson:16]: Doesn't the bullet "Verify SW_CMD_REQ/SW_CMD_STS registers/bits behave as predicted" cover this?

**Review by rasmus-madsen — APPROVED** [review]
I know this is a draft - but it looks ready to go to me

**Pull request merged**